src/ui/util/processLiveGameEvents.basketball.tsx

```diff
@@ -22,16 +21,73 @@ let playersByPid:
 const getName = (pid: number) => {
 	return playersByPid?.[pid]?.name ?? "???";
 };
+type StatRecordedType = {
+	type: (typeof statRecordedTypes)[number];
+	pid: number;
+	pidAst: number | undefined;
+	t: 0 | 1;
+};
+// We use this type for display on the logger
+const statRecordedTypes = [
+	"fgLowPost",
+	"fgLowPostAndOne",
+	"fgMidRange",
+	"fgMidRangeAndOne",
+	"tp",
+	"tpAndOne",
+	"fgTipIn",
+	"fgTipInAndOne",
+	"fgAtRim",
+	"fgAtRimAndOne",
+	"ft",
+	"fgPutBack",
+	"fgPutBackAndOne",
+	"blkAtRim",
+	"blkLowPost",
+	"blkMidRange",
+	"blkTp",
+	"blkTipIn",
+	"blkPutBack",
+	"pfNonShooting",
+	"pfAndOne",
+	"drb",
+	"orb",
+	"tov",
+] as const;
 
 export const getText = (
 	event: PlayByPlayEvent,
 	boxScore: {
 		numPeriods: number;
-		teams: [{ pts: number }, { pts: number }];
+		teams: [{ pts: number; players: any[] }, { pts: number; players: any[] }];
 	},
 ) => {
 	let texts: ReactNode[] | undefined;
 	let weights;
+	let playerStats: any;
+	let assistPlayerStats: any;
+	let playerTovStats: any;
+	if (!playersByPid) {
+		throw new Error("playersByPid is undefined");
+	} else if (statRecordedTypes.find((type) => type === event.type)) {
+		const statEvent = event as StatRecordedType; // Can cast as there is a type guard above
+		playerStats = playersByPid[statEvent.pid];
+		if (statEvent.pidAst !== undefined) {
+			assistPlayerStats = playersByPid[statEvent.pidAst];
+		}
+		// Now that we have the player stats, we can use them to generate the text
+	} else if (
+		event.type === "pfFG" ||
+		event.type === "pfAndOne" ||
+		event.type === "pfBonus" ||
+		event.type === "pfNonShooting" ||
+		event.type === "pfTP"
+	) {
+		playerStats = playersByPid[event.pid];
+	} else if (event.type === "stl") {
+		playerStats = playersByPid[event.pid];
+		playerTovStats = playersByPid[event.pidTov];
+	}
```

Comment:
I meant to delete even more! `playerStats`, `assistPlayerStats`, and `playerTovStats` can all be deleted. Just look up the player you need in `playersByPid` right when you need it. Simpler code, more efficient, and it works with TypeScript rather than fighting it.